### PR TITLE
Add and Remove group from user principals.

### PR DESCRIPTION
**Pull request by Natim**

### CODE REVIEW

**Comment by leplatrem** [comment]
I really wonder how you had the courage to write all this without tests! ;))

File: kinto/views/groups.py [hunk 1]
```diff
@@ -29,3 +31,38 @@ def __init__(self, *args, **kwargs):
...
+        for group in body['items']:
+            for member in group['members']:
+                group_id = '%s/%s' % (get_object_id(self.request), group)
+                permission.remove_user_principal(
+                    self.request.authenticated_userid,
+                    group_id)
+        return body
+
+    def process_record(self, new, old=None):
+        if old is not None:
```
Comment (leplatrem) [inline kinto/views/groups.py]: I believe it's the contrary :)

File: kinto/views/groups.py [hunk 2]
```diff
@@ -29,3 +31,38 @@ def __init__(self, *args, **kwargs):
...
+            # Add the group to the member principal
+            group_id = get_object_id(self.request)
+            permission.add_user_principal(user_id, group_id)
+
+        for member in removed_members:
+            # Remove the group from the member principal
+            group_id = get_object_id(self.request)
+            permission.remove_user_principal(user_id, group_id)
+
+        return new
```
Comment (leplatrem) [inline kinto/views/groups.py]: I think that should happen after the storage update was performed. 

Even if using ``process_record()`` is more elegant, I guess that with the existing code (and no event system), we would have to override every endpoint post/put/patch in order to run this code after the storage operation was executed :|

Comment (Natim) [inline kinto/views/groups.py]: Yes I did that first but then I decided to use process_record instead. In either way if one of the two fails we will be out of sync.

**Comment by ametaireau** [comment]
See #57 instead.

**Pull request closed without merging**